test/unit/transactionBodyTest.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-env mocha */
+
+import { retrieveTransactionBody } from '../../src/middleware/transactionBody'
+import uuid from 'uuid'
+import mongodb from 'mongodb'
+import { config } from '../../src/config'
+import fs from 'fs'
+
+describe('retrieveTransactionBody()', async () => {
+  let db
+  let client
+  const mongoClient = mongodb.MongoClient
+
+  before(async () => {
+    client = await mongoClient.connect(config.mongo.url)
+    db = client.db()
+  })
+
+  after(() => {
```

Comment:
@bradsawadye Your async test functions need to accept a callback parameter or return a promise. The current functions will complete before the callbacks have been called which may end up interfering with other tests and causing random failures like we are currently seeing on Travis. See https://mochajs.org/#asynchronous-code.